lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1866,7 +1866,7 @@ private function doMerge($entity, array &$visited, $prevManagedCopy = null, $ass
                 }
             }
 
-            if ($class->isVersioned) {
+            if ($class->isVersioned && !($this->isNotInitializedProxy($managedCopy) || $this->isNotInitializedProxy($entity))) {
```

Comment:
I feel it may be clearer to rewrite the conditional as:

    $class->isVersioned &&  !$this->isNotInitializedProxy($managedCopy) && !$this->isNotInitializedProxy($entity)